### DISCUSSION TITLE
Question about currency calculator

### DISCUSSION THREAD

**Discussion by starguo1998** [D3nan/Tafe24S1T102_Ext#3] [Q&A] [answered]
Hi all 
As I am do the currency calculator, but how can we structure the C# code to separate concerns and promote maintainability, considering the user interface (XAML) and backend logic?

**Comment by D3nan** [comment, marked as answer]
We will probably each have a XAML page for our own respective calculators, with their own C# code to operate that calculator.

**Reply by starguo1998** [reply]
hi Denan
Thanks for raising this important point. Separating concerns and promoting maintainability is crucial for building scalable and maintainable applications. In additon, we may slao can try the principles of the Model-View-ViewModel design pattern. This pattern helps separate the user interface (XAML) from the backend logic, making the codebase easier to understand, test, and maintain.
Xiaorui
